smarthash.spec

```diff
@@ -0,0 +1,48 @@
+# -*- mode: python ; coding: utf-8 -*-
```

Comment:
This file is autogenerated, and I don't think it should be checked in - is there a reason for this?

Reply:
It is auto generated if you want but you can also run

```pyinstaller [file].spec```

I did that in this case because there are a couple of files that I try to move in the case of the linux build since pyinstaller won't find them and they aren't installed in the vanilla install (checked by using a linux docker image, desktop may be different.)  Also it allows us to set the data and other command line options in the file rather than trying to keep a shell script updated also.

We can also remove the *.bat files by using pyinstaller for windows on a windows box, but I left them here in case you had other plans for them.

Reply:
Checking in a file like this is bad practise, because it ties the contents of the repository to a specific version of your OS.

You should instead use pyinstaller's --add-binary command to include additional binaries which it cannot find, local to your system.

Reply:
I removed the binaries, which weren't needed anyway as that is now handled by the hook if they are present in the pymediainfo python lib directory.  

I still think the spec file is a better way to specify the data directories needed.  As mentioned in the first bulleted scenario [here](https://pyinstaller.readthedocs.io/en/stable/spec-files.html).

This allows us to not have to have separate flows just because of the path separator difference in the add-data command line parameter.

If you still want me to remove this, I will as it's your project in the end.

Reply:
I'd prefer this was removed. Additional data directories can be specified on the command line with --add-data. This stuff can then be handled by an external, platform-specific build script.

Reply:
I did as was asked and added two build scripts.